### ISSUE TITLE
Any benchmark for the latest released model?

### ISSUE THREAD

**Issue by ControlNet** [KichangKim/DeepDanbooru#61]
Hi,

I love your project. Could you please provide some benchmark (accuracy, f1, etc) about the latest pretrained model in the release?

That will be a great help because I also want to train some models (EfficientNet, ViT, etc) by myself. By comparing the benchmark scores, I could understand the performance of the model better.

**Comment by koke2c95** [comment]
@ControlNet  remind you, danbooru tagger not suitable for trainers

the benchmark pretty useless, don't waste your valuable time

we already trained nfnet, regnet, ConvNeXt on danbooru2020

those too bad and, tagger is useless .... we don't know what this trained model can do

you should checkout this paper, what's next ["Transfer Learning for Pose Estimation of Illustrated Characters", Chen & Zwicker 2021](https://arxiv.org/abs/2108.01819)

but downstream task experiment show us, too bad even not good as  ....

we need VLM, text-image pretrained (like CLIP things) on LAION anime subset (remove tons anime uncorrelated datas)

then we need open vocab detector and a well captioner

then we can enter anime storytelling things

**Comment by KichangKim** [comment]
I don't have any benchmark test/score of DeepDanbooru for latest model.

**Comment by Daniel8811** [comment]
@ControlNet 
Try this: https://github.com/lucidrains/x-clip
You can use a pretrained text encoder and only train the image encoder. The only thing is that you will need to turn danbooru tags into sentences that can be interpreted by the pretrained text encoder  first.

**Comment by koke2c95** [comment]
hi @Daniel8811 
Thanks, I know there are already many variants of CLIP and [codebase available](https://github.com/dmarx/Multi-Modal-Comparators/issues/2) (including this one), but we still think it's better to use the single file provided by a kind guys

We think these common tokenizers are not suitable for anime stuff like bert, gpt2 tokenizer, [openai tokenizer](https://beta.openai.com/tokenizer)

We are also not to use common pre-trained text models, which are inefficient on anime noise keywords or booru-like

even the large amount of fandom alt text, describe scene is surprisingly poor, because western pre-training dataset does not know/care about the anime creator, character, work name, resulting in useless pre-training

[dall-e 2; aren't usually known by their name among Western audiences](https://www.reddit.com/r/dalle2/comments/uixrgg/comment/i7ihwl2/)

[4chan comments; Literally just an insufficient image dataset problem.....](https://boards.4chan.org/ic/thread/6080069/so-this-is-the-true-power-of-dalle#6080253)

We plan, included use booru's system, booru-like site featured images collection "pools", and structured "wiki" (or use pixiv dic), might can pretrained a good text model

lack aesthetic concept we use art wip stories, howto tutorial, artist's commentary, commission caption-image like pixiv, fanbox, skeb.jp, clipstudio , medibang, ichi-up, en tutorial

lack terms we use sankaku (its actually has a nice tag database ( types grouped ) than other booru-like (danbooru tags database very unified, non-unified noised like sankaku/gelbooru much better) , or hydrus PTR

As a pre-training dataset, the lack of these can lead to various problem when doing generation validation

We found training from scratch mostly bad like textures messly, wrong colored and lack coherence, the result show that a considerable amount of high quality photo of people, high quality real photo material for the actual used in creation inneeded, and long tail-media issues such as line sketch/monochrome also critical.

**Comment by ControlNet** [comment]
@koke2c95 Thank you for your reply. 

I decide to do simple auto-tagging (multi-label classification) with more lightweight and more accurate. So image generation or translation is not in my plan yet.

You said the label is very low quality. I fully understand it as these tags are community-driven, so the noise cannot be avoided. I'm thinking if it's possible to employ some self supervise learning and weak supervise learning techniques to improve it. 

Also, Is there any tag-based anime image dataset with accurate labels? 

BTW, regarding the height-width ratio of these images varies very significantly, I doubt naive resizing may not extract the features well. Using sliding window might be a better choice.

> I don't have any benchmark test/score of DeepDanbooru for latest model.

Thank you for your reply. 

@Daniel8811 Thank you for your suggestions. I know CLIP is an amazing work, and it's robust for unseen data.  But I'm not familiar with that. If the pretrained text encoder is used, I highly doubt the anime-style labels (yuri, genshin_impact, blue_hair) can be predicted well.

**Comment by koke2c95** [comment]
@Daniel8811

the answer is can't, openai CLIP's deal with anime stuff vision and text are weak

We ready going use [openclip VITB32 laion2b](https://github.com/mlfoundations/open_clip) for processing datas, the performance better on anime art-style recognition, medias, open vocab things and so on

and I don't like he gived labels :((

about write-up, sorry, we're plan not public any details until we successful

above comments is just our preliminary project proporsal, so far we've come far, a lot of knowledge, sources and datas

why, ? you know, the public is very aggressive on anime things and the copyright owners are extreme stupid

**Comment by ControlNet** [comment]
@Daniel8811 
> it contains less abstract concepts (like yuri) and more tags that are obvious to everyone (like blue hair)

Yes, it's possible although manually finding these "obvious" tags for thousands of tags are tricky.
